### PR TITLE
refactor: create Flag Enum

### PR DESCRIPTION
**Pull request by BrayanDSO**
Part of #14949

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/Flags.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    FLAG_ORANGE,
+    FLAG_GREEN,
+    FLAG_BLUE,
+    FLAG_PINK,
+    FLAG_TURQUOISE,
+    FLAG_PURPLE
+)
+annotation class Flag
+
+fun getFlagDrawable(@Flag flag: Int): Int {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/Flags.kt:42]: This is a new method

File: AnkiDroid/src/main/java/com/ichi2/anki/Flags.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    FLAG_NONE,
+    FLAG_RED,
+    FLAG_ORANGE,
+    FLAG_GREEN,
+    FLAG_BLUE,
+    FLAG_PINK,
+    FLAG_TURQUOISE,
+    FLAG_PURPLE
+)
+annotation class Flag
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/Flags.kt:40]: Renamed FlagDef to Flag

**Review by david-allison — APPROVED** [review]
Change is good

I'd personally prefer to move to an enum. The use of `IntDef` feels like a micro-optimisation which is no longer necessary

**Comment by BrayanDSO** [comment]
Stopped being lazy and did the Enum. PTAL

**Review by david-allison — APPROVED** [review]
Beautiful

**Review by david-allison — CHANGES_REQUESTED** [review]
Test failures: can't `%d` the enum in a Timber.

**Review by david-allison — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
somehow ended up being net-negative code even with that gigantic copyright notice
nice!

**Pull request merged**